recipes/pulseaudio/all/conanfile.py

```diff
@@ -0,0 +1,129 @@
+            raise ConanInvalidConfiguration("pulseaudio supports only linux currently")
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+
+    def requirements(self):
+        self.requires("libsndfile/1.0.30")
+        self.requires("libcap/2.45")
+        if self.options.with_alsa:
+            self.requires("libalsa/1.1.9")
+        if self.options.with_glib:
+            self.requires("glib/2.64.0")
+        if self.options.with_fftw:
+            self.requires("fftw/3.3.8")
+        if self.options.with_x11:
+            self.requires("xorg/system")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1h")   
+        # FIXME: enable when #2147 is merged
+        # if self.options.with_dbus
+        #     self.requires("dbus/1.12.16")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+            args=[]
+            
+            # FIXME: add dbus when #2147 is merged
+            for lib in ["alsa", "x11", "openssl"]:
+                args.append("--%s-%s" % ("enable" if getattr(self.options, "with_" + lib) else "disable", lib))
+            args.append("--%s-glib2" % ("enable" if self.options.with_glib else "disable"))
+            args.append("--%s-fftw" % ("with" if self.options.with_fftw else "without"))
+            if self.options.shared:
+                args.extend(["--enable-shared=yes", "--enable-static=no"])
+            else:
+                args.extend(["--enable-shared=no", "--enable-static=yes"])
+            args.append("--with-udev-rules-dir=%s" % os.path.join(self.package_folder, "bin", "udev", "rules.d"))
+            with tools.environment_append({"PKG_CONFIG_PATH": self.build_folder}):
+                with tools.environment_append({
+                        "FFTW_CFLAGS": tools.PkgConfig("fftw").cflags,
+                        "FFTW_LIBS": tools.PkgConfig("fftw").libs} if self.options.with_fftw else tools.no_op()):
+                    with tools.environment_append(RunEnvironment(self).vars):
+                        self._autotools.configure(args=args,  configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        autotools = self._configure_autotools()
+        autotools.install()
+        tools.rmdir(os.path.join(self.package_folder, "etc"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```

Comment:
Does the autotools project also install cmake scripts?
It's contagious, so it seems.
Is the name of the cmake_find_package generator correct?

Reply:
[yes](https://github.com/pulseaudio/pulseaudio/blob/master/PulseAudioConfig.cmake.in)
Unfortunately it cannot be regenerated by conan, the variables names (` PULSEAUDIO_INCLUDE_DIR`  and `PULSEAUDIO_LIBRARY`) don't match what conan generates (`<PKG-NAME>_INCLUDE_DIRS` and `<PKG-NAME>_LIBRARIES`)

Reply:
More bullets for https://github.com/conan-io/conan/issues/7691 :gun:

Reply:
Maybe add a FIXME about missing PULSEAUDIO_INCLUDE_DIR and PULSEAUDIO_LIBRARY?